Website/src/rendering/src/components/Products/ProductDetails.tsx

```diff
@@ -1,10 +1,36 @@
 import Head from 'next/head';
 import { useRouter } from 'next/router';
 import { useEffect } from 'react';
+import Skeleton from 'react-loading-skeleton';
 import useOcProductDetail from '../../hooks/useOcProductDetail';
 import { logViewEvent } from '../../services/CdpService';
 import ProductDetailsContent from './ProductDetailsContent';
 
+const ProductDetailsSkeleton = (): JSX.Element => {
```

Comment:
Thanks for the feedback Jeff, and I totally understand your concern around maintenance. What I'm noticing as I am trying to implement your suggestion however is that your strategy would work quite well when doing updates on an existing entity but it is increasing complexity quite a bit if trying to do inline as you suggest on first page load because no data is present at that time. 

Two issues I've ran into:
1. Need to add nullish coalescing operators in many places when accessing nested properties on props. These properties were previously safe to access because we could assume the value would be defined but now may not be if the component is loading
2. We would need to mock data to create a complete looking loader. For example on product detail the only thing that would render essentially would be the product name so it would look quite bare. For other things like line item list, the whole list would be missing without mocking the list line item data and then of course we need to make sure that mock works all the way down the component tree.

I'm fine continuing down the suggested path and using mocked data but I thought it was worth mentioning to you at least before I dig in further.